packages/frontend/core/src/desktop/dialogs/setting/workspace-setting/storage/blob-management.tsx

```diff
@@ -0,0 +1,305 @@
+import { Button, Checkbox, Loading, useConfirmModal } from '@affine/component';
+import { Pagination } from '@affine/component/member-components';
+import { BlobManagementService } from '@affine/core/modules/blob-management/services';
+import { useI18n } from '@affine/i18n';
+import type { ListedBlobRecord } from '@affine/nbstore';
+import { getAttachmentFileIcon } from '@blocksuite/affine/blocks';
+import { DeleteIcon } from '@blocksuite/icons/rc';
+import { useLiveData, useService } from '@toeverything/infra';
+import bytes from 'bytes';
+import { fileTypeFromBuffer } from 'file-type';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import * as styles from './style.css';
+
+const Empty = () => {
+  const t = useI18n();
+  return (
+    <div className={styles.empty}>
+      {t['com.affine.settings.workspace.storage.unused-blobs.empty']()}
+    </div>
+  );
+};
+
+const useBlob = (blobRecord: ListedBlobRecord) => {
+  const unusedBlobsEntity = useService(BlobManagementService).unusedBlobs;
+  const [url, setUrl] = useState<string | null>(null);
+  const [extension, setExtension] = useState<string | null>(null);
+  const [mimeType, setMimeType] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    let cancel = false;
+    let _url: string | null = null;
+    if (!blobRecord.key) return;
+
+    (async () => {
+      setLoading(true);
+      // hmm, maybe not load large files?
+      const blob = await unusedBlobsEntity.getBlob(blobRecord.key);
+      if (!blob || cancel) return;
+      const fileType = await fileTypeFromBuffer(await blob.arrayBuffer());
```

Comment:
can these async logic be moved to effect?

Reply:
I feel like using effect makes resource revoking a bit hassle. Changed the implementation with `useDisposable`. What do you think?